### ISSUE TITLE
Envoy 1.31.2 not generating cluster.outbound metrics

### ISSUE THREAD

**Issue by rwincewicz** [envoyproxy/envoy#36565]
*Title*: *Envoy 1.31.2 not generating `cluster.outbound` metrics*

*Description*:
I am seeing that the `cluster.outbound` metrics that used to be generated by Envoy are now missing having upgraded from 1.30.6 to 1.31.2. The Envoy upgrade has come from upgrading Istio from 1.22.5 to 1.23.3.

*Repro steps*:

Deploy a container in Kubernetes with an Istio sidecar. Istio is deployed via helm charts.

Query the Envoy metrics for the pod using `kubectl exec -n <NAMESPACE> <POD_NAME> -c istio-proxy -- pilot-agent request GET stats`
Observe that in Envoy version 1.30.6 there were metrics for `cluster.outbound`
```
cluster.outbound|33627||remote-service-33627.com.bind_errors: 0
cluster.outbound|33627||remote-service-33627.com.ssl.connection_error: 0
cluster.outbound|33627||remote-service-33627.com.ssl.fail_verify_error: 0
cluster.outbound|33627||remote-service-33627.com.ssl.fail_verify_no_cert: 0
cluster.outbound|33627||remote-service-33627.com.ssl.fail_verify_san: 0
cluster.outbound|33627||remote-service-33627.com.upstream_cx_protocol_error: 0
```
In Envoy version 1.31.2 these metrics do not appear despite the whole deployment being exactly the same. The only difference being the version of Istio and Envoy.

*Admin and Stats Output*:
/stats - [envoy-stats.txt](https://github.com/user-attachments/files/17361675/envoy-stats.txt)
/clusters - [envoy-clusters.txt](https://github.com/user-attachments/files/17361659/envoy-clusters.txt)
/server_info - [envoy-server_info.txt](https://github.com/user-attachments/files/17361679/envoy-server_info.txt)

*Config*:
[envoy-rev.json](https://github.com/user-attachments/files/17361722/envoy-rev.json)

*Logs*:
I don't have access to the separate Envoy logs but these are the logs from the istio-proxy sidecar container that contains the Envoy logs.
[envoy-log.txt](https://github.com/user-attachments/files/17361786/envoy-log.txt)

*Call Stack*:
N/A

**Comment by tyxia** [comment]
@rwincewicz  we have just released 1.32. I am wondering if you still see this issue with latest release

**Comment by rwincewicz** [comment]
We've retested our scenario with the latest version of Istio proxy (1.24.0) which includes version 1.32.2 of Envoy (reference to Envoy commit is [here](https://github.com/istio/proxy/blob/1.24.0/WORKSPACE#L26) and the metrics are still missing. Is there anything we can try with configuration to fix this issue or any further information you need to be able to debug this?
